core-bundle/tests/Contao/DcTableTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Contao;
+
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\Database;
+use Contao\Database\Result;
+use Contao\Database\Statement;
+use Contao\DC_Table;
+
+class DcTableTest extends TestCase
+{
+    /**
+     * @dataProvider getPalette
+     */
+    public function testGetPalette(array $dca, array $row, string $expected): void
+    {
+        $result = new Result([$row], '');
+
+        $statement = $this->createMock(Statement::class);
+        $statement
+            ->method('limit')
+            ->willReturn($statement)
+        ;
+
+        $statement
+            ->method('execute')
+            ->willReturn($result)
+        ;
+
+        $database = $this->createMock(Database::class);
+        $database
+            ->method('prepare')
+            ->willReturn($statement)
+        ;
+
+        $reflection = new \ReflectionClass(DC_Table::class);
+        $dataContainer = $reflection->newInstanceWithoutConstructor();
+
+        $method = $reflection->getMethod('import');
+        $method->setAccessible(true);
+        $method->invoke($dataContainer, $database, 'Database');
```

Comment:
This did fix the test, however, if you run `composer run unit-tests -- --filter DcTableTest`, there is a new issue:

<img width="922" alt="" src="https://user-images.githubusercontent.com/1192057/190340089-90b0376f-937f-446a-894d-fd861a90c495.png">

Maybe we should revert the fix and add `#[AllowDynamicProperties]` to the `DC_Table` class? Or we could adjust the `getPalette()` method so it uses `System::getContainer()->get('database_connection')` instead of `$this->Database`?

Reply:
How about b30c8500eabd407fa5f7eb6e4e8eb661e1323451 ?

Reply:
> How about https://github.com/contao/contao/commit/b30c8500eabd407fa5f7eb6e4e8eb661e1323451 ?

IMHO that's the right fix to get test isolation. :+1: If this is still not enough, we could also create a `DC_TableStub` (or an inline class) that removes the constructor and overrides `getCurrentRecord()`.